File: software/examples/javascript/ExampleSimple.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your DMX Bricklet

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var dmx = new Tinkerforge.BrickletDMX(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function (connectReason) {
        // Configure Bricklet as DMX master
        dmx.setDMXMode(Tinkerforge.BrickletDMX.DMX_MODE_MASTER);

        // Write DMX frame with 3 channels
        dmx.writeFrame([255, 128, 0]);
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{dmx_bricklet::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your DMX Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let dmx = DmxBricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Configure Bricklet as DMX master
    dmx.set_dmx_mode(DMX_BRICKLET_DMX_MODE_MASTER);

    // Write DMX frame with 3 channels
    dmx.write_frame(&[255, 128, 0])?;

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
